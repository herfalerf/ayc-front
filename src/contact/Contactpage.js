import React from "react";
import ContactForm from "./ContactForm";
import { makeStyles } from "@material-ui/core/styles";
import { Typography, Grid, Icon, Link } from "@material-ui/core";
import ic_contact_mail from "../common/images/ic_contact_mail.svg";
import ic_contact_phone from "../common/images/ic_contact_phone.svg";

// Contact Page
//
// Displays contact form
//
// Routed at /contact
//
// Routes -> Contact
//

const useStyles = makeStyles((theme) => ({
  root: { flexGrow: 1 },
  upper: {
    backgroundImage: `url('${process.env.PUBLIC_URL}/images/contact/contact01.png')`,
    backgroundSize: "cover",
    backgroundPosition: "center bottom",
    paddingTop: "27%",
    width: "100%",
    height: "100%",
  },
  logo: {
    maxHeight: 40,
    margin: theme.spacing(2),
  },
  mainIcon: {
    height: 100,
  },
  aboutBox: {
    padding: theme.spacing(4),
  },
  aboutItem: {
    padding: theme.spacing(2),
  },
  form: {
    padding: theme.spacing(4),
    backgroundColor: "rgb(244,245,248)",
  },
  message: {
    color: "#00c7d7",
  },
  contactBox: { padding: theme.spacing(4) },
  contactTitle: {
    padding: theme.spacing(1),
    color: "#00c7d7",
    fontWeight: "bold",
  },
  contactSubtitle: {
    paddingLeft: theme.spacing(1),
  },
  contactText: {
    padding: theme.spacing(2),
  },
}));

function Contactpage({ addCustomer }) {
  console.debug("Contactpage", "addCustomer=", typeof addCustomer);

  const classes = useStyles();

  return (
    <div>
      <div className={classes.upper}></div>
      <Grid container justifyContent="center" className={classes.aboutBox}>
        <Grid item xs={12}>
          <Typography align="center">
            <img
              src={process.env.PUBLIC_URL + "/images/logos/Logo03a.png"}
              alt="AYC Logo"
              className={classes.logo}
            ></img>
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Typography variant="h4" align="center">
            It Starts With a Conversation
          </Typography>
        </Grid>
        <Grid item xs={12} sm={8} md={6}>
          <Typography variant="subtitle1" align="center">
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Lorem
            ipsum dolor sit amet, consectetur adipisicing elit. Libero at
            voluptate officiis sed atque, ipsam quod est dicta veritatis eaque
            aut, sequi sapiente veniam eveniet aliquid totam?
          </Typography>
        </Grid>
        <Grid
          container
          justifyContent="center"
          item
          className={classes.contactBox}
        >
          <Grid container alignItems="center" item xs={12} sm={4} lg={3}>
            <Grid item>
              <Typography align="center">
                <Link component="a" href="tel:1231231234">
                  <Icon>
                    <img
                      src={ic_contact_phone}
                      alt="phone icon"
                      className={classes.mainIcon}
                    />
                  </Icon>
                </Link>
              </Typography>
            </Grid>
            <Grid item className={classes.contactText}>
              <Typography variant="subtitle2" className={classes.contactTitle}>
                CALL US
              </Typography>
              <Typography className={classes.contactSubtitle}>
                123-123-1234
              </Typography>
            </Grid>
          </Grid>
          <Grid container alignItems="center" item xs={12} sm={4} lg={3}>
            <Grid item>
              <Typography align="center">
                <Link component="a" href="mailto:test@email.com">
                  {" "}
                  <Icon>
                    <img
                      src={ic_contact_mail}
                      alt="email icon"
                      className={classes.mainIcon}
                    />
                  </Icon>
                </Link>
              </Typography>
            </Grid>
            <Grid item className={classes.contactText}>
              <Typography variant="subtitle2" className={classes.contactTitle}>
                EMAIL US
              </Typography>
              <Typography className={classes.contactSubtitle}>
                test@email.com
              </Typography>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
      <Grid container justifyContent="center" className={classes.form}>
        <Grid item>
          <Typography variant="h5" className={classes.message}>
            Send a Message
          </Typography>
          <ContactForm addCustomer={addCustomer} />
        </Grid>
      </Grid>
    </div>
  );
}

export default Contactpage;
