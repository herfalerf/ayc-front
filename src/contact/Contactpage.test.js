import React from "react";
import { render } from "@testing-library/react";
import Contactpage from "./Contactpage";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <Contactpage />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <Contactpage />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
