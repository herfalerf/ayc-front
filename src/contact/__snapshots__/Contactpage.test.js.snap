// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-upper-17"
    />
    <div
      class="MuiGrid-root makeStyles-aboutBox-20 MuiGrid-container MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
        >
          <img
            alt="AYC Logo"
            class="makeStyles-logo-18"
            src="/images/logos/Logo03a.png"
          />
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
        >
          It Starts With a Conversation
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-6"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter"
        >
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Libero at voluptate officiis sed atque, ipsam quod est dicta veritatis eaque aut, sequi sapiente veniam eveniet aliquid totam?
        </h6>
      </div>
      <div
        class="MuiGrid-root makeStyles-contactBox-24 MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-lg-3"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="tel:1231231234"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root"
                >
                  <img
                    alt="phone icon"
                    class="makeStyles-mainIcon-19"
                    src="ic_contact_phone.svg"
                  />
                </span>
              </a>
            </p>
          </div>
          <div
            class="MuiGrid-root makeStyles-contactText-27 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root makeStyles-contactTitle-25 MuiTypography-subtitle2"
            >
              CALL US
            </h6>
            <p
              class="MuiTypography-root makeStyles-contactSubtitle-26 MuiTypography-body1"
            >
              123-123-1234
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-lg-3"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="mailto:test@email.com"
              >
                 
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root"
                >
                  <img
                    alt="email icon"
                    class="makeStyles-mainIcon-19"
                    src="ic_contact_mail.svg"
                  />
                </span>
              </a>
            </p>
          </div>
          <div
            class="MuiGrid-root makeStyles-contactText-27 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root makeStyles-contactTitle-25 MuiTypography-subtitle2"
            >
              EMAIL US
            </h6>
            <p
              class="MuiTypography-root makeStyles-contactSubtitle-26 MuiTypography-body1"
            >
              test@email.com
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-form-22 MuiGrid-container MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h5
          class="MuiTypography-root makeStyles-message-23 MuiTypography-h5"
        >
          Send a Message
        </h5>
        <div>
          <form
            action="#"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
              >
                name*
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="name"
                  type="name"
                  value=""
                />
              </div>
            </div>
            <br />
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
              >
                email*
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="email"
                  type="email"
                  value=""
                />
              </div>
            </div>
            <br />
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
              >
                company
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="company"
                  type="company"
                  value=""
                />
              </div>
            </div>
            <br />
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
              >
                phone
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="phone"
                  type="phone"
                  value=""
                />
              </div>
            </div>
            <br />
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
              >
                message
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                  name="message"
                  rows="1"
                  style="height: 0px; overflow: hidden;"
                />
                <textarea
                  aria-hidden="true"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                  readonly=""
                  style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                  tabindex="-1"
                />
              </div>
            </div>
            <br />
            <br />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-linkButton-29 MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Send A Message
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
